webui/react/src/components/EditableMetadata.tsx

```diff
@@ -0,0 +1,108 @@
+import { Button, Dropdown, Form, Input, Menu } from 'antd';
+import { FormListFieldData } from 'antd/lib/form/FormList';
+import React, { useCallback, useMemo } from 'react';
+
+import css from './EditableMetadata.module.scss';
+import Icon from './Icon';
+import InfoBox, { InfoRow } from './InfoBox';
+
+interface Props {
+  editing: boolean;
```

Comment:
**suggestion:** if editing is most likely going to be `false` and an opt in by the user, I would suggest making it optional, so that not supplying one would default to `undefined` which would be the same as an explicit `false` behavior.
```
editing?: boolean;
```